src/lib/eth-calls.ts

```diff
@@ -1,21 +1,17 @@
 import { ContractReceipt } from "ethers";
 /* eslint-disable-next-line camelcase */
 import { TablelandTables__factory } from "@tableland/eth";
-import { Connection } from "../interfaces.js";
-import { contractAddresses } from "./util.js";
+import { Connection } from "./connection.js";
+import { getSigner } from "./util.js";
 
 async function registerTable(
   this: Connection,
   query: string
 ): Promise<ContractReceipt> {
-  const signer = this.signer;
+  const signer = this.signer ?? (await getSigner());
```

Comment:
lazily get signer if not provided. this will throw in node if a signer isn't provided